### PR TITLE
Documentation: The good self-review

### LINKED ISSUE #1426: [Documentation] Write "The good: self-review"-section for contribution guidelines
**This is part of a larger issue - please read: https://github.com/kirbydesign/designsystem/issues/1393#issuecomment-802766408 before continuing!**

**Short description**

A "The good: self-review" section should be added to the contribution guidelines.  An outline already exists on the branch [`documentation/contribution-guidelines`](https://github.com/kirbydesign/designsystem/tree/documentation/contribution-guidelines) which should be expanded upon.

### PR DESCRIPTION
**Pull request by jkaltoft**
This PR closes #1426

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement (to existing content)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?

Issue Number: #[replace with issue number]

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -531,12 +531,33 @@ For more on good git commits, see (among _many_ other):
 * [Turbo Git Commit Message Standard](https://gist.github.com/turbo/efb8d57c145e00dc38907f9526b60f17)
 * [Check Out These 5 Git Tips Before Your Next Commit](https://medium.com/walmartglobaltech/check-out-these-5-git-tips-before-your-next-commit-c1c7a5ae34d1)
 
-### Self-review 
+### The good self-review
```
Comment (MadsBuchmann) [inline .github/CONTRIBUTING.md:534]: I agree with what we talked about regarding _the good_ "wrapper" section, so I think we should name them all as you've done here. But let us do that in the end. 

Suggested formatting, feel free to ignore it:
```suggestion
### The Good: Self-Review
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -531,12 +531,33 @@ For more on good git commits, see (among _many_ other):
...
-  + for example don't reorder imports in files that is not directly related to the issue at hand
-* Check for unused imports
-* Other self review points go here
+A lot of comments and requests for change during review of a pull request can sometimes feel demotivating. The good self-review can save yourself and the reviewer from frustrations during peer review.
+
+The code review process has a better chance of going smooth if you conduct a self-review before opening the pull request and marking it _Ready for review_. Removing gravel in the road helps keeping code review focused and unnecessary discussions can be avoided.
+
+#### Unrelated changes
+
+Avoid doing unrelated changes in files that are not directly part of your pull-request. For example, don't submit changes like formatting or reordering imports if they're the _only_ changes made in those files.
```
Comment (MadsBuchmann) [inline .github/CONTRIBUTING.md:542]: 🐜-size suggestion: 
```suggestion
Avoid unrelated changes in files that are not directly part of your pull-request. For example, don't submit changes like formatting or reordering imports if they're the _only_ changes made in those files.
```

**Review by MadsBuchmann — COMMENTED** [review]
Besides the comments - I think it would be neat to change the headings to be more actionable e.g.: 
- Do not include unrelated changes
- Make sure files are formatted/prettified
- Look out for linting warnings  
- Delete temporary changes

Also has the advantage of the reader being able to read only the headings, to get the essence of the section 😄

File: .github/CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkaltoft) [reply .github/CONTRIBUTING.md:542]: > doe snot

😁

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -531,12 +531,33 @@ For more on good git commits, see (among _many_ other):
...
-### Self-review 
+### The good self-review
 
-* Don't change files that is not related to your pull-request
-  + for example don't reorder imports in files that is not directly related to the issue at hand
-* Check for unused imports
-* Other self review points go here
+A lot of comments and requests for change during review of a pull request can sometimes feel demotivating. The good self-review can save yourself and the reviewer from frustrations during peer review.
+
+The code review process has a better chance of going smooth if you conduct a self-review before opening the pull request and marking it _Ready for review_. Removing gravel in the road helps keeping code review focused and unnecessary discussions can be avoided.
```
[the remark this answers is not in the document]
Comment (jkaltoft) [reply .github/CONTRIBUTING.md:538]: I like your words better than mine 😊

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -531,12 +531,33 @@ For more on good git commits, see (among _many_ other):
...
+
+#### Formatting and linting
+
+Use Prettier to format the files you're changing. If you are editing a file that for some reason has not already been properly formatted, you'll probably see a lot of changes when you diff the file after you format it. Try to put the formatting changes in a separate commit if possible e.g., by creating a _formatting commit_ before committing your other changes. It will be easier for the reviewer to see your actual changes if they have the option to ignore the formatting changes.
+
+Use the linting rules to sort imports and remove unused variables, imports etc.
+
+#### Temporary changes
+
+Look for temporary changes you have used during development, like `console.log()` statements, TODO comments and code that has been commented out. They do not belong in the codebase and should be removed. TODOs should either be implemented right away or you should create a new issue for solving it later.
```
[the remark this answers is not in the document]
Comment (jkaltoft) [reply .github/CONTRIBUTING.md:552]: I like the "Dead code" suggestion, but I think I'll leave it out for now. It's not necessarily temporary changes in the same  way the other three are. Maybe I can merge it somewhere else 😊

**Review by MadsBuchmann — APPROVED** [review]
LGTM! 🚀

**Pull request merged**